sirepo/package_data/static/js/sirepo.js

```diff
@@ -1672,8 +1675,8 @@ SIREPO.app.factory('panelState', function(appState, requestSender, simulationQue
             return progressText + ' ' + new Array(count % 3 + 2).join('.');
         }
         return appState.isAnimationModelName(name)
-            ? 'Requesting Data'
-            : 'Waiting';
+        ? 'Requesting Data'
+        : 'Waiting';
```

Comment:
Not personally a fan of this formatting (esp. since this will fit comfortably on one line), but the indent should be consistent. See above:
```
var delay = isPlaying && ! isHidden
            ? 1000 / parseInt(appState.models[modelName].framesPerSecond || 2)
            : 0;
```

Reply:
1000 and 2 are magic numbers. They should come from somewhere or be documented. The parseInt should be this:

```js
appState.models[modelName].framesPerSecond ? parseInt(appState.models[modelName].framesPerSecond) : 2
```

Maybe this should be encapsulated in a function so it is properly documented. Then the 1000 and 2 could be consts in the function and it would be clearer what they mean.

Reply:
@mkeilman I can def change that indent.

@robnagler I am not totally clear on the meaning behind the 1000 and the 2 myself. I can wrap the ternary expression in a function if you like. The line you wrote above however would change the meaning though wouldn't it? 

Would something like this be good?
```javascript
const getDelay = () => {
    const x = 1000;
    const y = 2;
    if (isPlaying && ! isHidden) {
        return  x/(parseInt(appState.models[modelName].framesPerSecond || y)
    }
    return 0;
}
var delay = getDelay()
```
(again, I am unclear on the semantics of 1000 and 2

Reply:
@robnagler are you saying framePeriod would be a better name?

Reply:
I think renaming the constants x and y isn't anymore helpful than having them in place, so give the function and constants a long meaningful name. Also, make sure everything fits with 100 character lines for the new coding style.

```javascript
const millisecondsUntilNextFrameRequest = () => {
    if (isPlaying && ! isHidden) {
        const secondsToMilliseconds = 1000;
        const defaultFramesPerSecond = 2;
        return secondsToMilliseconds / (
            parseInt(appState.models[modelName].framesPerSecond
                || defaultFramesPerSecond
            )
    }
    return 0;
}
var delay = getMillisecondsUntilNextFrameRequest()
```

Reply:
I agree with @moellep on x and y. @gurhar1133 I asked you to update DesignHints: https://github.com/radiasoft/pykern/issues/187

I really don't like `parseInt(thing || default)`. defaultFramesPerSecond is already an int and it should be default milliseconds. Yes, I'm being pedantic:

```js
const millisecondsUntilNextFrameRequest = () => {
    if (! isPlaying || isHidden) {
        return 0;
    }
    const milliseconds = 1000;
    var x = appState.models[modelName].framesPerSecond; 
    if (! x) {
        return  .5 * milliseconds;
    }
    return  milliseconds / parseInt(x);
}
```

I don't think we need to document the default in this case, because "! x" clear indicates it is a default.

Reply:
The parseInt() is due to our current limitation with enums (framesPerSecond is an enum value) - it is always a string value mapped to a string description. This also makes booleans awkward because Boolean is an enum. So you see code like this in the python:
```python
    if v.controlSettings_readOnly == '0':
	properties.write = []
```
rather than:
```python
    if not v.controlSettings_readOnly:
	properties.write = []
```

Reply:
There is already an issue related to Boolean/enum #4167

Reply:
> The parseInt() is due to our current limitation with enums (framesPerSecond is an enum value)

You did not understand the point of my refactoring. parseInt for appState.models[modelName].framesPerSecond is necessary, and I reduced the code to be more readable (imiho) to just be the parseInt on that value, not on the number 2, which is already an integer so does not need to be parsed. 

@radiasoft/software please chime in on the readability issue.

Reply:
One other item related to our discussion on React yesterday: denormalized state on the stack. [This code](https://github.com/radiasoft/sirepo/blob/5d3d71b1cfb25ef30f71aad8bf1c6b1e59cee3aa/sirepo/package_data/static/js/sirepo.js#L1316) is using denormalized model state unnecessarily:

```js
                   let endTime = new Date().getTime();
                    let elapsed = endTime - frameRequestTime;
                    if (elapsed < delay) {
                        $interval(
                            function() {
                                callback(index, data);
                            },
                            delay - elapsed,
                            1
                        );
                    }
                    else {
                        callback(index, data);
                    }
```

There's no reason to cache delay. Rather, you should ask at that point of needing it, that is, `delay` should be computed at the time that the callback is received from the server, not before. 

Furthermore, millisecondsUntilNextFrameRequest is not the right name. It really is framePeriod (in milliseconds, which is irrelevant). The reason the callback is delayed is that we want a frame rate of 2 frames per second (or whatever the particular model specifies).

I would write the code this way (ignore the indentation

```js
const framePeriod = () => {
    if (! isPlaying || isHidden) {
        return 0;
    }
    const milliseconds = 1000;
    var x = appState.models[modelName].framesPerSecond;
    if (! x) {
        return .5 * milliseconds;
    }
    return milliseconds / parseInt(x);
}

let e = framePeriod() - (new Date().getTime() - frameRequestTime);
if (e <= 0) {
    callback(index, data);
    return;
}
$interval(
    function() {
        callback(index, data);
    },
    e,
    1,
);
```

Why is this important? If a user clicks stop, then isPlaying and isHidden have already been cached in delay, and they were checked at frameRequestTime instead of now (btw, new Date().getTime() should be now() or something).